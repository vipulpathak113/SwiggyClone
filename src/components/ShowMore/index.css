.show-more {
    position: relative;
    background-color: #fff;
    flex: 0 0 30%;
    height: 292px;
    display: flex;
    top: 30px;
    justify-content: center;
    align-items: center;
    border: 3px solid #e46d47;
    transform: scale(0.8) translate(-92px, -25px);
}

.show-more-title {
    font-size: 24px;
    color:#e46d47;
    cursor: pointer;
    font-weight: 600;
}

