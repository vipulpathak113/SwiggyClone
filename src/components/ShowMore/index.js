import React from "react";

import "./index.css";

const ShowMore = ({ remainingItems, showMore }) => {
  // calling fncn using props
  let showMorehandler = () => {
    showMore();
  };
  return (
    <div className="show-more">
      <div className="show-more-title" onClick={showMorehandler}>
        <span href="">+{remainingItems} More</span>
      </div>
    </div>
  );
};
export default ShowMore;
