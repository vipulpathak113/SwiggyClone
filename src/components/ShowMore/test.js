import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import ShowMore from './index.js'

describe('<ShowMore /> spec', () => {
    it('renders the component', () => {
        let container = render(<ShowMore/>)
        expect(container.firstChild).toMatchSnapshot()
    })
    it('assert there are 12 Show More Tiles', () => {
        render(<ShowMore remainingItems="12"/>)
 
        expect(document.querySelectorAll('.show-more-title span').length).toBe(1)
        expect(document.querySelector('.show-more-title span').innerHTML).toBe("+12 More")
    })
})
