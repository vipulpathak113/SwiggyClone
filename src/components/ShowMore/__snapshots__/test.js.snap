// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowMore /> spec renders the component 1`] = `undefined`;
